### PR TITLE
VDP-2: added java generator

### PR DESCRIPTION
**Pull request by sksadjad**
added java generator to pex-openapi project. 
Since java generator for `openapi-generator-maven-plugin` doesn't support `oneOf` added a small tweak to generator configs to put `object` when encountering oneof

### CODE REVIEW

**Review by mehmetbicak — COMMENTED** [review]
2 questions  :

The description says ”Ensure documentation is updated” I did not see any documentation update. Should we missed the documentation change? 

Is there any action in-progress/done related to publishing to nexus repo? “Publish on our open-source-snapshots/releases nexus repo.“

File: docs/release-pex-models.md [hunk 1]
```diff
@@ -49,3 +49,13 @@ yarn publish --access public
 Check on `npmjs.com`
 
 Install the module following the included `README.md`. 
+
+
+## Releasing java
+
+If you have the correct settings in your local workspace, you can easily publish a new version on Sphereon repository (`sphereon-opensource-snapshots`)
```
Comment (mehmetbicak) [inline docs/release-pex-models.md:56]: What is the correct settings for the local workspace? Can you also add some detail for this?

**Review by mehmetbicak — CHANGES_REQUESTED** [review]
Related to java release: 
What is the correct settings for the local workspace? Can you also add some detail for this?

File: docs/release-pex-models.md (hunk 1 above)
Comment (sksadjad) [reply docs/release-pex-models.md:56]: I'm not sure what I should put here, you have to have correct settings.xml in your maven home to be able to connect, you have to have the correct credentials

**Comment by nklomp** [comment]
Please only mention settings.xml authentication. 

Let's not start explaining to us internally how you use maven

File: docs/release-pex-models.md (hunk 1 above)
Comment (nklomp) [reply docs/release-pex-models.md:56]: Yeah agree. Let's not explain to us internally how you use maven

**Review by nklomp — APPROVED** [review]

**Comment by nklomp** [comment]
Approved based upon jira comment

**Review by mehmetbicak — APPROVED** [review]

**Pull request merged**